src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      return Promise.reject(`Error: ${response.status}`);
+    }
+    return response.json();
+  }
+
+  getAllData() {
+    return Promise.all([this.loadUserInfo(), this.getInitialCards()])
+      .then(([user, cards]) => {
+        return { user, cards };
+      })
+      .catch((error) => console.error("Error loading data:", error));
+  }
+
+  loadUserInfo() {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "GET",
+      headers: this.headers,
+    })
+      .then(this._checkResponse)
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.